packages/format-library/src/text-color/index.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { useCallback, useMemo, useState } from '@wordpress/element';
+import { RichTextToolbarButton } from '@wordpress/block-editor';
+import { Dashicon } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { default as InlineColorUI, getActiveColor } from './inline';
+
+const name = 'core/text-color';
+const title = __( 'Text Color' );
+
+const EMPTY_ARRAY = [];
+
+function TextColorEdit( { value, onChange, isActive, activeAttributes } ) {
+	const colors = useSelect( ( select ) => {
+		const { getSettings } = select( 'core/block-editor' );
+		return get( getSettings(), [ 'colors' ], EMPTY_ARRAY );
+	} );
+	const [ isAddingColor, setIsAddingColor ] = useState( false );
+	const enableIsAddingColor = useCallback( () => setIsAddingColor( true ), [
+		setIsAddingColor,
+	] );
+	const disableIsAddingColor = useCallback( () => setIsAddingColor( false ), [
+		setIsAddingColor,
+	] );
+	const colorIndicatorStyle = useMemo( () => {
+		const activeColor = getActiveColor( name, value, colors );
+		if ( ! activeColor ) {
+			return undefined;
+		}
+		return {
+			backgroundColor: activeColor,
+		};
+	}, [ value, colors ] );
+	return (
+		<>
+			<RichTextToolbarButton
+				className="format-library-text-color-button"
+				name={ isActive ? 'text-color' : undefined }
+				icon={
+					<>
+						<Dashicon icon="editor-textcolor" />
+						{ isActive && (
+							<span
+								className="format-library-text-color-button__indicator"
+								style={ colorIndicatorStyle }
+							/>
+						) }
+					</>
+				}
+				title={ title }
+				onClick={ enableIsAddingColor }
+			/>
+			{ isAddingColor && (
+				<InlineColorUI
+					name={ name }
+					addingColor={ isAddingColor }
+					onClose={ disableIsAddingColor }
+					isActive={ isActive }
+					activeAttributes={ activeAttributes }
+					value={ value }
+					onChange={ onChange }
+				/>
+			) }
+		</>
+	);
+}
+
+export const textColor = {
+	name,
+	title,
+	tagName: 'span',
+	className: 'has-inline-color',
```

Comment:
I was hoping a class wouldn't be needed when we do this:
https://github.com/WordPress/gutenberg/issues/15478

But I don't think it should block this feature. Let's work that out later.